Loot/Collectible.gd

```diff
@@ -13,20 +13,35 @@ signal collected
 
 var _is_flying_to_player := false
 var _is_collected := false
+var _animation_delta = 0
 var value: int
+var velocity: Vector3
+
+const DAMPING_FACTOR = 0.01
 
 var _player
 
 func _ready() -> void:
 	$ItemArea.connect("body_entered", self, "_on_body_entered")
+	var initial_fly_direction = randf() * PI * 2
+	velocity = Vector3(cos(initial_fly_direction) * 15, 2, sin(initial_fly_direction) * 15)
 
 func set_value(value: int) -> void:
 	self.value = value
 
 func _physics_process(delta: float) -> void:
 	if _is_flying_to_player:
 		var distance_to_player = (_player.transform.origin - transform.origin).length()
-		move_and_slide((_player.transform.origin - transform.origin).normalized() / pow(distance_to_player, 2) * 100)
+		velocity += (_player.transform.origin - transform.origin).normalized() / pow(distance_to_player, 2) * 50
+	velocity *= pow(DAMPING_FACTOR, delta)
+	move_and_slide(velocity)
+	_animate_bounce(delta)
+
+func _animate_bounce(delta: float) -> void:
```

Comment:
Should we call this `_animate_float` or just `_float`, because bounce feels misleading when I see this for the first time.

Reply:
Oh can. Cos I was also stuck on how to name it

Reply:
Refactored to _animate_hover